src/modules/battery.rs

```diff
@@ -75,27 +75,76 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 
 fn get_battery_status() -> Option<BatteryStatus> {
     let battery_manager = battery::Manager::new().ok()?;
-    match battery_manager.batteries().ok()?.next() {
-        Some(Ok(battery)) => {
-            log::debug!("Battery found: {:?}", battery);
-            let battery_status = BatteryStatus {
-                percentage: battery.state_of_charge().value * 100.0,
-                state: battery.state(),
-            };
+    let batteries = battery_manager.batteries().ok()?;
+    let battery_contructor = batteries
+        .filter_map(|battery| match battery {
+            Ok(battery) => {
+                log::debug!("Battery found: {:?}", battery);
+                Some(BatteryContructor {
+                    energy: battery.energy().value,
+                    energy_full: battery.energy_full().value,
+                    state: battery.state(),
+                })
+            }
+            Err(e) => {
+                log::debug!("Unable to access battery information:\n{}", &e);
+                None
+            }
+        })
+        .fold(
+            BatteryContructor {
+                energy: 0.0,
+                energy_full: 0.0,
+                state: battery::State::Unknown,
+            },
+            |mut acc, x| {
+                acc.energy += x.energy;
+                acc.energy_full += x.energy_full;
+                acc.state = battery_state_fold(acc.state, x.state);
+                acc
+            },
+        );
+    if battery_contructor.energy_full != 0.0 {
```

Comment:
Do we need this condition?
`energy_full` could be `0.0` in the case of a depleted battery that is charging. I think we should still show the charging status in that case.

Reply:
We need to consider the case in which either the battery manager or the subsequent filter_map does not return any battery information. In those cases, the fold operation will return a `BatteryContructor` with both `energy` and `energy_full` at 0.0.
This will end up in a division by zero on line 109, but apart for that, we should return None in those situations.
I agree with you that this is not a very good condition, but I think that we need something like that.

Reply:
My mistake. I thought that `energy_full` was `energy`.
Yeah, this is a fine condition to have. 👍